### ISSUE TITLE
UnLua vs. sluaunreal

### ISSUE THREAD

**Issue by TheWitheredStriker** [Tencent/UnLua#324]
Hello! Hope you're having a great day.

I saw that, in addition to this repository, you guys apparently have another repository to embed Lua into Unreal: https://github.com/Tencent/sluaunreal

I think both of them look incredibly interesting and I would love to get started with them, but I must honestly admit that I don't immediately see the difference between the two.

Where do UnLua and sluaunreal differ? I'd be curious to see how they compare to each other.

Thanks in advance!

**Comment by Eros-L** [comment]
> 
> Hello! Hope you're having a great day.
> 
> I saw that, in addition to this repository, you guys apparently have another repository to embed Lua into Unreal: https://github.com/Tencent/sluaunreal
> 
> I think both of them look incredibly interesting and I would love to get started with them, but I must honestly admit that I don't immediately see the difference between the two.
> 
> Where do UnLua and sluaunreal differ? I'd be curious to see how they compare to each other.
> 
> Thanks in advance!

@TheWitheredStriker Same question here! Since the document is incomplete on feature list and "not so english-friendly", I found it difficult to (neither) tell the difference between UnLua and sluaunreal (nor single out one of them) for beginners. Then, I spent time seeking additional resources and FINALLY came to the following conclusions. Hope these make sense to you.

1. They BOTH allow access to reflected fields (e.g. `UCLASS`, `UPROPERTY`, `UFUNCTION`, `UENUM`, etc.) from Lua script.
2. They BOTH provide extension methods for exporting non-reflected fields to Lua script.
3. UnLua allows binding Lua script to a gameplay object in run-time; sluaunreal only allows binding statically through the class constructor.
4. UnLua overrides blueprint events and introduces the `Overridden` namespace for access to the overriddens; sluaunreal simply reimplements blueprint events.
5. UnLua handles Lua hot reloading natively; sluaunreal provides useful tools such as debugger and profiler.

There are still too many to mention. Really I've tried both of them in my project and observed that UnLua has better performance in most cases. Btw, maintenance of sluaunreal's repository has already been discontinued afaik while UnLua's not. Hence I would personally suggest that you look into UnLua. :-)
